### PR TITLE
Update to v2.1.1 and the Terraform Plugin SDK

### PR DESCRIPTION
**Pull request by jen20**
This PR updates to v2.1.1 of the upstream Terraform provider, and replaces the dependency on Terraform with a dependency on the new SDK. This is a draft for now (to see how CI works out), and likely _does not_ want merging as is, instead it needs to depend on the outcome of pulumi/pulumi-terraform#498.

The dependency graph is a _lot_ more sane now...

### CODE REVIEW

**Comment by stack72** [comment]
Fails on go1.13 as we have a dep that changes between go versions - I am fine with this as we need to make a decision on go1.13.x soon!

File: sdk/nodejs/randomPassword.ts [hunk 1]
```diff
@@ -19,11 +25,16 @@ import * as utilities from "./utilities";
...
+ *     overrideSpecial: "_%@",
  *     special: true,
  * });
  * const example = new aws.rds.Instance("example", {
- *     password: password.result,
+ *     allocatedStorage: 64,
+ *     engine: "mysql",
+ *     instanceClass: "db.t3.micro",
+ *     password: "",
+ *     "random_string.password.result": [{}],
```
Comment (pgavlin) [inline sdk/nodejs/randomPassword.ts]: This is very strange: what does the source example look like?

Comment (jen20) [reply sdk/nodejs/randomPassword.ts]:
```hcl
resource "random_password" "password" {
  length = 16
  special = true
  override_special = "_%@"
}

resource "aws_db_instance" "example" {
  instance_class = "db.t3.micro"
  allocated_storage = 64
  engine = "mysql"
  username = "someone"
  password = random_string.password.result
}
```

Comment (jen20) [reply sdk/nodejs/randomPassword.ts]: I'd say this looks like a bug in tf2pulumi, @pgavlin?

Comment (stack72) [reply sdk/nodejs/randomPassword.ts]: Their example is not correct

```
resource "random_password" "password" {
  length = 16
  special = true
  override_special = "_%@"
}

resource "aws_db_instance" "example" {
  instance_class = "db.t3.micro"
  allocated_storage = 64
  engine = "mysql"
  username = "someone"
  password = random_string.password.result
}
```

Notice it isn't using the `${random_password.password.result}`

Comment (jen20) [reply sdk/nodejs/randomPassword.ts]: https://github.com/terraform-providers/terraform-provider-random/pull/77 is partly related here @pgavlin - the source example is wrong - but our handling of it is probably not ideal either.

Comment (pgavlin) [reply sdk/nodejs/randomPassword.ts]: I think that this is a `tf2pulumi` issue. Here's what happens when I try to generate this locally:
```
hela:t pgavlin$ tf2pulumi >index.ts
error: creating tree module: Error parsing main.tf: At 12:14: Unknown token: 12:14 IDENT random_string.password.result
```

Pretty sure that this is valid HCL2 syntax, and that's what `tf2pulumi` is choking on.

Comment (stack72) [reply sdk/nodejs/randomPassword.ts]: @pgavlin the password uses `random_string` but the resource is `random_password`

Comment (pgavlin) [reply sdk/nodejs/randomPassword.ts]: Yeah, but that's a semantic error, not a syntax error. `tf2pulumi` can't even parse the file when I run it locally. `tf2pulumi` can tolerate the semantic error; we'd just generate something that references the unknown resource.

**Pull request merged**